met-web/src/components/engagement/view/EmailPanel.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { useState } from 'react';
+import {
+    Grid,
+    Typography,
+    Checkbox,
+    Button,
+    TextField,
+    FormControl,
+    FormControlLabel,
+    FormHelperText,
+} from '@mui/material';
+import { EmailPanelProps } from './types';
+
+const style = {
+    position: 'absolute',
+    top: '50%',
+    left: '50%',
+    transform: 'translate(-50%, -50%)',
+    width: 700,
+    bgcolor: 'background.paper',
+    boxShadow: 10,
+    pt: 2,
+    px: 4,
+    pb: 3,
+    height: 700,
+};
+
+const label = { inputProps: { 'aria-label': 'Checkbox demo' } };
+
+const EmailPanel = ({ email, checkEmail, handleClose, updateEmail }: EmailPanelProps) => {
+    const [checked, setChecked] = useState(false);
+    const [emailFormError, setEmailFormError] = useState({
+        terms: false,
+        email: false,
+    });
+
+    const validateForm = () => {
+        const errors = {
+            terms: !checked,
+            email: email === '',
```

Comment:
!email also works for email === ''. Boolean('') is considered false. So !email will work here

Reply:
that's cool ill change that then